sccp/figures/figureS2.py

```diff
@@ -1,133 +1,150 @@
-#lupus_genes.to_csv('fold_change.rnk', index=False, header=False, sep='\t')
-
-
-############################################################################################
-# SEAN MAKES SOME SCATTERPLOTS ########
-############################################################################################
-
-# resetting the combo_lupus association because here I want all observational things to map more variables to color etc
-
-lupus_pan = lupus_data.to_df()
-
-lupus_observations = lupus_data.obs
-
-combo_lupus = lupus_observations.merge(lupus_pan, 
-                                       how= "left",
-                                       left_index=True,
-                                       right_index=True,
-                                       validate="one_to_one")
-
-
-# get average by donor_id
-
-mean_gene_dictionary = {}
-# add all the genes, and tell it to calc mean for each
-for gene in lupus_genes.index.tolist():
-    mean_gene_dictionary[gene] = "mean"
-
-avg_combo_lupus = combo_lupus.groupby(["ind_cov"]).agg(mean_gene_dictionary)
-
-# want to combine with col abt pop_cov and SLE status
-obs_cov_status = lupus_observations[["ind_cov", "SLE_status", "pop_cov"]].drop_duplicates(subset="ind_cov")
-
-
-lupus_by_patient = pd.merge(obs_cov_status, avg_combo_lupus, right_index=True, left_on="ind_cov")
+from .common import subplotLabel, getSetup
+import matplotlib.pyplot as plt
+import scipy.cluster.hierarchy as sch
+import tensorly as tl
+
+
+####################################################################################
+## LOAD IN DATA -- CAN BE MOVED TO imports/scRNA 
+####################################################################################
+
+def load_lupus_data(third_axis= "ind_cov", n_rand = 100):
+    """Import Lupus PBMC dataset."""
+    X = anndata.read_h5ad("/opt/andrew/lupus/lupus.h5ad")
+    
+    # reorder X so that all of the patients are in alphanumeric order. this is important
+    # so that we can steal cell typings at this point
+    obsV = X.obs_vector('ind_cov')
+    sgUnique, sgIndex = np.unique(obsV, return_inverse=True)
+
+    ann_data_objects = [X[sgIndex == sgi, :] for sgi in range(len(sgUnique))]
+
+    X = anndata.concat(ann_data_objects, axis = 0)
+
+    # select n_rand random patients; evenly split between lupus and SLE
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    # REMOVE THIS SECTION TO HAVE ALL INDIVIDUALS INCLUDED (and n_rand arg in function def)
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    np.random.seed(42)
+    SLE_patients = X.obs["ind_cov"][X.obs["SLE_status"] == "SLE"].to_numpy()
+    healthy_patients = X.obs["ind_cov"][X.obs["SLE_status"] == "Healthy"].to_numpy()
+    random_patients = np.random.choice(SLE_patients, int(n_rand/2), replace=False).tolist() + np.random.choice(healthy_patients, int(n_rand/2), replace=False).tolist()
+
+    X = X[X.obs['ind_cov'].isin(random_patients), :]
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
```

Comment:
@aarmey this can be removed alongside the `n_rand` keyword argument to run the full dataset